shared/src/main/scala/pl/metastack/metatime/DateTime.scala

```diff
@@ -1,18 +1,87 @@
 package pl.metastack.metatime
 
-trait DateTime {
-  def date: Date = ???
-  def format: String = ???
+trait DateTime extends Time with Date {
   def format(fmt: String): String = ???
   def timestamp: Timestamp = ???
-  def -(time: Component): DateTime = ???
-  def +(time: Component): DateTime = ???
-  def -(dateTime: DateTime): DateTime = ???
-  def +(dateTime: DateTime): DateTime = ???
-  def fromNow: Offset[DateTime] = ???
+
+  override def equals(that: Any): Boolean =
+    that match {
+      case other: DateTime => other.year == year && other.month == month && other.day == day &&
+        h == other.h && m == other.m && s == other.s && ms == other.ms
+      case _ => false
+    }
+
+  override def unix(): Unix = Unix(
+    Year(year).unix().value +
+    Month(month).unix().value +
+    Day(day).unix().value +
+    Hour(h).unix().value +
+    Minute(m).unix().value +
+    Second(s).unix().value +
+    Millisecond(ms).unix().value)
 }
 
 object DateTime {
-  def now(): DateTime = ???
-  def apply(year: Int, month: Int, day: Int): DateTime = ???
+
+  def calculateDateTime(milliDate: Long, milliTime: Long): DateTime = {
+    val date = Date(milliDate)
+    val time = Time(milliTime)
+    DateTime(date.year, date.month, date.day, time.h, time.m, time.s, time.ms).asInstanceOf[DateTime]
+  }
+
+  def now(): DateTime = calculateDateTime(Date.now().unix().value, Time.now.unix().value)
+  def apply(millisecond: Long): DateTime = calculateDateTime(millisecond, millisecond)
+
+
+  def apply(yr: Int, mo: Int, dy: Int,
+            hour: Int, minute: Int, second: Int, millisecond: Int): DateTime =
+    new DateTime {
+      override val year: Int = yr
+      override val month: Int = mo
+      override val day: Double = dy
+      override val h: Int = hour
+      override val m: Int = minute
+      override val s: Int = second
+      override val ms: Int = millisecond
+    }
+
+  def apply(yr: Int = 0,  mo: Int = 1, dy: Double = 1.0,
+            hr: Int = 0, min: Int = 0, sec: Int = 0,
+            milli: Int = 0): Date = new DateTime {
+    override val year: Int = yr
+    override val month: Int = mo
+    override val day: Double = dy
+    override val h: Int = hr
+    override val m: Int = min
+    override val s: Int = sec
+    override val ms: Int = milli
+  }
+
+  def apply(comp : Component) : DateTime = {
+    comp match {
+      case dT: DateTime =>
+        DateTime(dT.year, dT.month, dT.day, dT.h, dT.m, dT.s, dT.ms).asInstanceOf[DateTime]
```

Comment:
Absolute I agree but without it IntellijIDEA gives this error: "Expression of type Date doesn't conform to expected type DateTime". Any idea how can I get ride of this error?